hearth/lib/hearth/configuration.rb

```diff
@@ -7,5 +7,13 @@ def initialize(**options)
       Hearth::Config::Resolver.resolve(self, options, self.class.defaults)
       validate!
     end
+
+    def dup
```

Comment:
This is better I think. Also I think you can use each_with_object here:

```
members.each_with_object(super) do |member, copy|
  copy[member] = self[member].dup
end
```

Reply:
nice! Thats cleaner